src/queries/UpdateElementQueries.ts

```diff
@@ -0,0 +1,98 @@
+import {ProjectElements, ProjectSettings, Schemes, VocabularyElements} from "../config/Variables";
+import {qb} from "./QueryBuilder";
+import {DELETE, INSERT} from "@tpluscode/sparql-builder";
+
+export function updateProjectElement(del: boolean, ...ids: string[]): string {
+	let data: { [key: string]: string[] } = {};
+	let deletes: string[] = [];
+	let inserts: string[] = [];
+
+	if (ids.length === 0) return "";
+	for (let id of ids) {
+		checkElem(id);
+		const iri = ProjectElements[id].iri;
+		const iriDiagram = `${ProjectElements[id].iri}/diagram`;
+		const vocabElem = VocabularyElements[ProjectElements[id].iri];
+		const scheme = vocabElem.inScheme;
+		const graph = Schemes[vocabElem.inScheme].graph;
+		const types = vocabElem.types.map(type => qb.i(type));
+		const labels = Object.keys(vocabElem.labels).filter(lang => vocabElem.labels[lang]).map(lang => qb.ll(vocabElem.labels[lang], lang));
+		const altLabels = vocabElem.altLabels.map(alt => qb.ll(alt.label, alt.language));
+		const definitions = Object.keys(vocabElem.definitions).filter(lang => vocabElem.definitions[lang]).map(lang => qb.ll(vocabElem.definitions[lang], lang));
+		const names = Object.keys(ProjectElements[id].selectedLabel).map(lang => qb.ll(ProjectElements[id].selectedLabel[lang], lang));
+		const diagrams = ProjectElements[id].diagrams.map((diag) => qb.i(`${ProjectElements[id].iri}/diagram-${diag + 1}`));
+
+		if (!(graph in data)) data[graph] = [];
+		if (del) data[graph].push(
+			qb.s(qb.i(iri), 'rdf:type', qb.a(types)),
+			qb.s(qb.i(iri), 'skos:prefLabel', qb.a(labels)),
+			qb.s(qb.i(iri), 'skos:altLabel', qb.a(altLabels), altLabels.length > 0),
+			qb.s(qb.i(iri), 'skos:definition', qb.a(definitions), definitions.length > 0),
+			qb.s(qb.i(iri), 'skos:inScheme', qb.i(scheme)),
+			qb.s(qb.i(scheme), 'skos:hasTopConcept', qb.i(iri), vocabElem.topConcept !== undefined)
+		);
+		data[graph].push(
+			qb.s(qb.i(iriDiagram), 'rdf:type', 'og:element'),
+			qb.s(qb.i(iriDiagram), 'og:context', qb.i(ProjectSettings.contextIRI)),
+			qb.s(qb.i(iriDiagram), 'og:id', qb.ll(id)),
+			qb.s(qb.i(iriDiagram), 'og:iri', qb.i(iri)),
+			qb.s(qb.i(iriDiagram), 'og:name', qb.a(names)),
+			qb.s(qb.i(iriDiagram), 'og:diagram', qb.a(diagrams)),
+			qb.s(qb.i(iriDiagram), 'og:active', qb.ll(ProjectElements[id].active)),
+		);
+
+		if (del) deletes.push(...[
+			qb.s(qb.i(iri), 'rdf:type', '?type'),
```

Comment:
again, lots of constants and repeated patterns

qb.s(qb.i(iriDiagram), PROP, qb.ll(VAL)),